pkg/resource/slo/delete.go

```diff
@@ -29,15 +29,23 @@ import (
 	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/delete/pointer"
 )
 
-type client interface {
+type source interface {
 	List(ctx context.Context) (api.PagedListResponse, error)
 	Delete(ctx context.Context, id string) (api.Response, error)
 }
 
-func Delete(ctx context.Context, c client, dps []pointer.DeletePointer) error {
+type Deleter struct {
```

Comment:
Question: would there be value in including an assertion that this implements `delete.Deleter`?

```
var _ delete.Deleter = Deleter{}
```